File: 697-degree-of-an-array/697-degree-of-an-array.py
class Solution:
    def findShortestSubArray(self, nums: List[int]) -> int:
        left = {}
        right = {}
        count = {}
        
        for i, x in enumerate(nums):
            
            if x not in left:
                left[x] = i
            
            right[x] = i
            count[x] = 1 + count.get(x, 0)
            
        ans = len(nums)
        degree = max(count.values())
        
        for x in count:
            
            if degree == count[x]:
                ans = min(ans, right[x] - left[x] + 1)
        return ans
                

File: 697-degree-of-an-array/697-degree-of-an-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findShortestSubArray = function(nums) {
    
    let left = {}
    let right = {}
    let count = {}
    
    for(let i = 0; i < nums.length; i++){
        
        if (left.hasOwnProperty([nums[i]])){
            
        }else{
            left[nums[i]] = i
            
        }
        right[nums[i]] = i
        
        if (count.hasOwnProperty([nums[i]])){
            count[nums[i]]++
            
        }else{
            count[nums[i]] = 1
            
        }
        
    }
    
    let deg_array = []
    for (let key in count){
        
        deg_array.push(count[key])
    }
    
    
    let res = nums.length
    
    let degree = Math.max(...deg_array)
    
    
    for(let key in count){
        
        if (count[key] === degree){
            res = Math.min(res, right[key] - left[key] + 1)
        }
    }
    
    return res
    
    
    
};